core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -379,7 +380,12 @@ public boolean tryComplete() {
                 partitionsAcquired.clear();
                 localPartitionsAlreadyFetched.clear();
             }
-            return forceComplete();
+            if (isCompleted()) {
+                return false;
+            } else {
+                forceComplete();
+                return true;
+            }
```

Comment:
If `forceComplete()` is called inside a try-catch block, we need to determine whether an exception was thrown during the call.
To avoid relying on the return value, we should use `isCompleted()` to check the completion state instead.

Reply:
Where is the exception coming from? Can forceComplete() throw an exception?

Reply:
`forceComplete()` calls `onComplete()`, and the test case `DelayedShareFetchTest#testTryCompleteLocksReleasedOnCompleteException` verifies the behavior when `onComplete()` throws an exception.
https://github.com/apache/kafka/blob/cc0f06554bacc9fe086c4f4c5ecee1b5dc75ae9b/core/src/test/java/kafka/server/share/DelayedShareFetchTest.java#L820